### PR TITLE
Refactoring + addition new methods

### PR DESCRIPTION
**Pull request by dosumis**
Warning - breaking change on get_images method!

@Robbie1977 @admclachlan - would like review on method & arg names & doc.

In particular - would like review of 

* get_neurons_downstream_of
* get_neurons_downstream_of
* get_connected_neurons_by_type
* get_images
* get_images_by_type

Is it clear what the methods do from names & doc.  Is it clear what args refer to?
Is the terminology consistent enough?  (probably not - this is a first pass)

(Note method formerly called get_images is now get_instances)

### CODE REVIEW

File: src/vfb_connect/cross_server_tools.py [hunk 1]
```diff
@@ -43,26 +71,137 @@ def get_terms_by_region(self, region, cells_only=False, verbose=False, query_by_
...
+
+    def get_neurons_downstream_of(self, neuron, threshold, classification=None, query_by_label=True):
+        """Get all neurons downstream of individual `neuron` (short_form if query_by_label=False, otherwise label)
+        with connection strength > threshold.  Optionally restrict target neurons to those specified by
+        `classification = 'class expression' e.g. "'Kenyon cell'" or "'neuron' that overlaps 'lateral horn'"."""
+        return self._get_neurons_connected_to(neuron=neuron, threshold=threshold, direction='upstream',
+                                              classification=classification, query_by_label=query_by_label)
+
+    def get_neurons_upstream_of(self, neuron, threshold, classification=None, query_by_label=True):
+        """Get all neurons downstream of individual `neuron` (short_form if query_by_label=False, otherwise label)
```
Comment (admclachlan) [inline src/vfb_connect/cross_server_tools.py:127]: upstream of?

File: src/vfb_connect/cross_server_tools.py [hunk 2]
```diff
@@ -43,26 +71,137 @@ def get_terms_by_region(self, region, cells_only=False, verbose=False, query_by_
...
+                        "downstream.short_form as target_neuron_id, downstream.label as target_neuron_name"
+        r = self.nc.commit_list([cypher_query])
+        dc = dict_cursor(r)
+        return pd.DataFrame.from_records(dc)
+
+    def get_neurons_downstream_of(self, neuron, threshold, classification=None, query_by_label=True):
+        """Get all neurons downstream of individual `neuron` (short_form if query_by_label=False, otherwise label)
+        with connection strength > threshold.  Optionally restrict target neurons to those specified by
+        `classification = 'class expression' e.g. "'Kenyon cell'" or "'neuron' that overlaps 'lateral horn'"."""
+        return self._get_neurons_connected_to(neuron=neuron, threshold=threshold, direction='upstream',
```
Comment (admclachlan) [inline src/vfb_connect/cross_server_tools.py:123]: Not sure on how direction works in_get_neurons_connected_to but is upstream right for downstream?

File: src/vfb_connect/cross_server_tools.py [hunk 3]
```diff
@@ -43,26 +71,137 @@ def get_terms_by_region(self, region, cells_only=False, verbose=False, query_by_
...
+                                              classification=classification, query_by_label=query_by_label)
+
+    def get_neurons_upstream_of(self, neuron, threshold, classification=None, query_by_label=True):
+        """Get all neurons downstream of individual `neuron` (short_form if query_by_label=False, otherwise label)
+         with connection strength > threshold.  Optionally restrict target neurons to those specified by
+         `classification = 'class expression' e.g. "'Kenyon cell'" or "'neuron' that overlaps 'lateral horn'"."""
+        return self._get_neurons_connected_to(neuron=neuron, threshold=threshold, direction='downstream',
+                                              classification=classification, query_by_label=query_by_label)
+
+    def get_connected_neurons_by_type(self, upstream_type, downstream_type, weight, query_by_label=True):
```
Comment (admclachlan) [inline src/vfb_connect/cross_server_tools.py:133]: What does weight do here? Same as threshold?

**Review by Robbie1977 — APPROVED** [review]

File: src/vfb_connect/cross_server_tools.py (hunk 3 above)
Comment (dosumis) [reply src/vfb_connect/cross_server_tools.py:133]: Yep.  Will change to weight.

**Pull request merged**